v2/features/brinery/BrineryTableRowDetails.tsx

```diff
@@ -0,0 +1,208 @@
+import { FC, Fragment } from "react";
+import { Disclosure, Transition } from "@headlessui/react";
+import { NULL_ADDRESS } from "picklefinance-core/lib/model/PickleModel";
+import { Trans, useTranslation } from "next-i18next";
+import { useWeb3React } from "@web3-react/core";
+import type { Web3Provider } from "@ethersproject/providers";
+import { Chains } from "picklefinance-core";
+
+import { roundToSignificantDigits, shortenAddress } from "v2/utils";
+import MoreInfo from "v2/components/MoreInfo";
+import Button from "v2/components/Button";
+import Link from "v2/components/Link";
+import { BrineryWithData } from "v2/store/core";
+import DetailsToggle from "../farms/DetailsToggle";
+import FarmDocs from "../farms/FarmDocs";
+
+interface Props {
+  brinery: BrineryWithData;
+  hideDescription?: boolean;
+}
+
+interface InfoProps {
+  label: string;
+  tooltipText: string | null;
+  value: string;
+}
+
+const InfoRowContent: FC<InfoProps> = ({ label, tooltipText, value }) => (
+  <p className="flex py-2 text-foreground-alt-200 justify-between">
+    <span className="font-body font-bold">
+      {label}
+      {tooltipText && (
+        <MoreInfo>
+          <span className="text-foreground-alt-200 text-sm">{tooltipText}</span>
+        </MoreInfo>
+      )}
+    </span>
+    <span className="font-medium text-primary text-base">{value}</span>
+  </p>
+);
+
+interface ComponentRowProps {
+  property: string;
+  value: string;
+}
+
+const ComponentRow: FC<ComponentRowProps> = ({ property, value }) => (
+  <p className="flex justify-between">
+    <span className="ml-4 font-body font-medium text-sm">{property}</span>
+    <span className="text-primary font-medium text-sm">{value}</span>
+  </p>
+);
+
+const BrineryTableRowDetails: FC<Props> = ({ brinery, hideDescription }) => {
+  const { t } = useTranslation("common");
+  const chain = Chains.get(brinery.chain);
+  const { library } = useWeb3React<Web3Provider>();
+
+  const totalTokensInBrinery = parseFloat(brinery.brineryBalance.tokens);
+
+  const metamaskAdd = async () => {
```

Comment:
Likewise, we should reuse this function and the components above

<img width="1499" alt="Screen Shot 2022-05-07 at 23 45 28" src="https://user-images.githubusercontent.com/7811733/167263862-738db52e-e688-4e5a-bbf6-35f9526f57df.png">